File: route_policy_cmp/src/lex/rpsl_object.rs
use super::*;

#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub struct AutNum {
    pub name: String,
    pub body: String,
    pub imports: Versions,
    pub exports: Versions,
}

#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub struct AsOrRouteSet {
    pub name: String,
    pub body: String,
    pub members: Vec<String>,
}

impl AsOrRouteSet {
    pub fn new(name: String, body: String, members: Vec<String>) -> Self {
        Self {
            name,
            body,
            members,
        }
    }
}

#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub struct PeeringSet {
    pub name: String,
    pub body: String,
    pub peerings: Vec<Peering>,
}

#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub struct FilterSet {
    pub name: String,
    pub body: String,
    pub filters: Vec<Filter>,
}

File: rpsl_lexer/src/rpsl_lexer/rpsl_object.py
from dataclasses import dataclass


@dataclass
class RPSLObject:
    closs: str
    name: str
    body: str


@dataclass
class AutNum:
    """aut-num class."""

    name: str
    body: str
    n_import: int
    n_export: int
    imports: dict[str, dict[str, list[dict]]]
    exports: dict[str, dict[str, list[dict]]]


@dataclass
class AsSet:
    """as-set class."""

    name: str
    body: str
    members: list[str]


@dataclass
class RouteSet:
    """route-set class."""

    name: str
    body: str
    members: list[str]
    """members and mp-members."""


@dataclass
class PeeringSet:
    """peering-set class."""

    name: str
    body: str
    peerings: list[dict]


@dataclass
class FilterSet:
    """filter-set class."""

    name: str
    body: str
    filters: list[dict]
